mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1822,8 +1822,13 @@ pub fn wait_for_output_spend(
     Box::new(fut.boxed().compat())
 }
 
-pub fn tx_enum_from_bytes(coin: &UtxoCoinFields, bytes: &[u8]) -> Result<TransactionEnum, String> {
-    let mut transaction: UtxoTx = try_s!(deserialize(bytes).map_err(|err| format!("{:?}", err)));
+pub fn tx_enum_from_bytes(coin: &UtxoCoinFields, bytes: &[u8]) -> Result<TransactionEnum, MmError<TransactionErr>> {
+    let mut transaction: UtxoTx = deserialize(bytes).map_to_mm(|e| TransactionErr::InvalidTx(e.to_string()))?;
+
+    if bytes.len() != transaction.tx_hex().len() {
+        return MmError::err(TransactionErr::InvalidTx("Transaction isn't valid.".to_string()));
```

Comment:
How about `format!("Expected '{}' lenght of the serialized transaction, found '{}'", bytes.len(), transaction.tx_hex().len())`?

Reply:
Done